### ISSUE TITLE
VOC algorithm

### ISSUE THREAD

**Issue by itsnils** [adafruit/Adafruit_CircuitPython_SGP40#2]
Hello I would like to measure the VOC value.
I have too little python i2c experience to realize this.

I found a project where the VOC algorithm is available.
[Link](https://github.com/agners/micropython-sgp40/blob/master/sgp40.py)

do you have soon for to insert this VOC algorithm?

thank you very much for your work

**Comment by ladyada** [comment]
yeah it is not possible at this time because @BoschSensortec has not release the library code in python only arduino!

**Comment by itsnils** [comment]
Hello @ladyada 
thank you for your answer. You probably meant @Sensirion AG
Too bad the VOC algorithm is still not implemented. I can't use the SGP40 sensor like this with Python.

**Comment by ladyada** [comment]
correct!

**Comment by CrackXT** [comment]
Hello everyone, @Sensirion has published the code n python, please have a look here. [https://github.com/Sensirion/python-i2c-sgp4x](url)

**Comment by AVanVlack** [comment]
@CrackXT
I cant seem to find an algorithm in that code base. It looks like it is only commands for communicating with the sensor and getting the raw data.  Unfortunately the same as this CircuitPython driver can do now.

I did find that DFRobot has a python algorithm and have a fork of this respiratory with it working on CircuitPython. [See code and examples here](https://github.com/AVanVlack/Adafruit_CircuitPython_SGP40)

I would make a pull request but not sure how Adafruit feels about an algorithm not directly from Sensation.

**Comment by ladyada** [comment]
sure any PR would be most welcome :)

**Comment by Curt-is-Pi** [comment]
Looks like the example for the compensated measurements has the '_index' and '_raw' mixed up.

compensated_raw_gas = sgp.measure__index_(temperature=temperature, relative_humidity=humidity

should read

compensated_raw_gas = sgp.measure_raw(temperature = temperature, relative_humidity = humidity)

And

voc_index = sgp.measure_raw(temperature=temperature, relative_humidity=humidity)

should read

voc_index = sgp.measure_index(temperature = temperature, relative_humidity = humidity)

Right?
